### PR TITLE
108268 new attachment endpoint vol2

### PR DESCRIPTION
**Pull request by janid1967**
[AB#108268](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/108268)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.Command/Validators/LabelValidator.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        var label = await GetLabelAsync(labelText, cancellationToken);
+
+        return label is not null;
+    }
+
+    public async Task<bool> IsVoidedAsync(string labelText, CancellationToken cancellationToken)
+    {
+        var label = await GetLabelAsync(labelText, cancellationToken);
+
+        return label is not null && label.IsVoided;
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/Validators/LabelValidator.cs:28]: If a label does not exist, does that mean it is voided?
From a mathematical/logical point of view I would say no.
From a practical point of view, it's your call...

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/LabelRepository.cs [hunk 2]
```diff
@@ -8,4 +13,7 @@ public LabelRepository(CompletionContext context)
         : base(context, context.Labels)
     {
     }
+
+    public Task<List<Label>> GetManyAsync(IEnumerable<string> labels, CancellationToken cancellationToken) =>
+        DefaultQuery.Where(l => labels.Contains(l.Text)).ToListAsync(cancellationToken);
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/LabelRepository.cs:18]: Should the comparison be case-insensitive or not?

File: src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs [hunk 3]
```diff
@@ -515,6 +514,37 @@ public async Task GetPunchItemAttachmentDownloadUrl_AsReader_ShouldGetUrl()
...
+
+        // Assert
+        AssertRowVersionChange(attachmentGuidAndRowVersion.RowVersion, newAttachmentRowVersion);
+
+        var _ = await PunchItemsControllerTestsHelper.GetPunchItemAttachmentsAsync(
+            UserType.Writer,
+            TestFactory.PlantWithAccess,
+            punchItemGuidAndRowVersion.Guid);
+
+        // todo Assert description and labels
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:545]: To be done in another pull request?

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.WebApi.IntegrationTests/PunchItems/PunchItemsControllerTests.cs:545]: yes, working on it ... PR Vol 3 for same PBI

File: src/Equinor.ProCoSys.Completion.Command/Validators/LabelValidator.cs (hunk 1 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/Validators/LabelValidator.cs:28]: No, but I don't agree that you can say so either.
If IsVoidedAsync == true, we know that the label exists AND it is voided and that is what we want to know

I don't see the problem and all other similar IsVoidedAsync in other validator-classes are exact the same

**Review by jsoi-eq — APPROVED** [review]

File: src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/LabelRepository.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Infrastructure/Repositories/LabelRepository.cs]: Please check latest commit @jsoi-eq

**Review by jsoi-eq — APPROVED** [review]
